vim_binding.js

```diff
@@ -119,6 +119,30 @@ define([
         }
       }
   };
+
+  // soft-failing shortcut helper - works around the hard-failing shortcut
+  // manager in IPython
+  var add_shortcuts = function add_shortcuts(manager_name, data){
+    var manager;
+    if(manager_name === "edit"){
+      manager = km.edit_shortcuts;
+    }else if(manager_name === "command"){
+      manager = km.command_shortcuts;
+    }else{
+      return console.error('invalid shortcut manager');
+    }
+    manager.clear_shortcuts();
```

Comment:
It would be better to extract this function out side of `add_shortcuts` while developer cannot recognize the function `add_shortcuts` actually clear shortcuts first.